### PR TITLE
Node inventory db history tests

### PR DESCRIPTION
**Pull request by besawn**
This pull request contains a test case that runs a suite of tests to confirm that node inventory items are being copied to DB history tables as expected.

Expected output when run on a Witherspoon node populated with all of the expected Coral hardware:
```
[root@c650mnp05 c650mnp05]# cat /test/results/buckets/basic/node_inventory_db_history.log
------------------------------------------------------------
          Starting Node Inventory DB History Bucket
------------------------------------------------------------
Mon Oct 15 15:17:39 EDT 2018
------------------------------------------------------------
SINGLE_COMPUTE is set to 'c650f99p18'
Configured DB tables for testing: csm_hca csm_gpu csm_node csm_processor_socket csm_dimm csm_ssd
------------------------------------------------------------
Test Case 0: Clean table data; confirm all tables are empty
PASS
------------------------------------------------------------
Test Case 1: Start the compute daemon; confirm inventory entries are created in the active and history tables.
PASS
------------------------------------------------------------
Test Case 2: Change the compute node state; confirm no entries are created in the history tables.
PASS
------------------------------------------------------------
Test Case 3: Restart the compute daemon; confirm no entries are created in the history tables.
PASS
------------------------------------------------------------
Test Case 4: Stop the compute daemon and delete it from the DB; confirm entries are created in the history tables.
PASS
------------------------------------------------------------
Test Case 5: Start the compute daemon; confirm active tables are populated and new history entries are created.
PASS
------------------------------------------------------------
Test Case 6: Change individual inventory fields and recollect inventory, confirm expected history is created.
PASS
------------------------------------------------------------
             Node Inventory DB History Bucket Passed
------------------------------------------------------------
Failed test cases:

------------------------------------------------------------
```

### CODE REVIEW

File: csmtest/buckets/basic/node_inventory_db_history.sh [hunk 1]
```diff
@@ -0,0 +1,655 @@
...
+   /bin/psql -U postgres csmdb -c "DELETE FROM ${history_tables[$tbl]} where node_name='$SINGLE_COMPUTE';" 1>> ${TEMP_LOG} 2>&1
+done
+
+read_db_after
+
+check_all_tables_empty
+check_return_exit $? 0
+
+rm -f ${TEMP_LOG}
+echo "------------------------------------------------------------" >> ${LOG}
```
Comment (pdlun92) [inline csmtest/buckets/basic/node_inventory_db_history.sh]: Removing these dividing lines would also be more consistent with formatting in different buckets
```suggestion
```

File: csmtest/buckets/basic/node_inventory_db_history.sh [hunk 2]
```diff
@@ -0,0 +1,655 @@
...
+check_all_tables_populated "Test Case 1"
+(( rc=rc+$? ))
+
+check_all_tables_new_history "Test Case 1"
+(( rc=rc+$? ))
+
+check_return_flag $rc "Test Case 1: Not all tables were populated with inventory data."
+
+rm -f ${TEMP_LOG}
+echo "------------------------------------------------------------" >> ${LOG}
```
Comment (pdlun92) [inline csmtest/buckets/basic/node_inventory_db_history.sh]: Removing these dividing lines would also be more consistent with formatting in different buckets
```suggestion
```

File: csmtest/buckets/basic/node_inventory_db_history.sh [hunk 3]
```diff
@@ -0,0 +1,655 @@
...
+
+${CSM_PATH}/csm_node_attributes_update -n "$SINGLE_COMPUTE" -s "IN_SERVICE" 1>> ${TEMP_LOG} 2>&1
+sleep 1
+
+read_db_after
+check_all_tables_no_new_history "Test Case 2"
+check_return_flag $? "Test Case 2: New history entries were created incorrectly."
+
+rm -f ${TEMP_LOG}
+echo "------------------------------------------------------------" >> ${LOG}
```
Comment (pdlun92) [inline csmtest/buckets/basic/node_inventory_db_history.sh]: Removing these dividing lines would also be more consistent with formatting in different buckets
```suggestion
```

File: csmtest/buckets/basic/node_inventory_db_history.sh [hunk 4]
```diff
@@ -0,0 +1,655 @@
...
+
+xdsh $SINGLE_COMPUTE "systemctl restart csmd-compute" 1>> ${TEMP_LOG} 2>&1
+sleep 1
+
+read_db_after
+check_all_tables_no_new_history "Test Case 3"
+check_return_flag $? "Test Case 3: New history entries were created incorrectly."
+
+rm -f ${TEMP_LOG}
+echo "------------------------------------------------------------" >> ${LOG}
```
Comment (pdlun92) [inline csmtest/buckets/basic/node_inventory_db_history.sh]: Removing these dividing lines would also be more consistent with formatting in different buckets
```suggestion
```

File: csmtest/buckets/basic/node_inventory_db_history.sh [hunk 5]
```diff
@@ -0,0 +1,655 @@
...
+check_all_active_tables_empty "Test Case 4"
+(( rc=rc+$? ))
+
+check_all_tables_new_history "Test Case 4"
+(( rc=rc+$? ))
+
+check_return_flag $rc "Test Case 4: Active tables are not empty or new history entries were not created."
+
+rm -f ${TEMP_LOG}
+echo "------------------------------------------------------------" >> ${LOG}
```
Comment (pdlun92) [inline csmtest/buckets/basic/node_inventory_db_history.sh]: Removing these dividing lines would also be more consistent with formatting in different buckets
```suggestion
```

File: csmtest/buckets/basic/node_inventory_db_history.sh [hunk 6]
```diff
@@ -0,0 +1,655 @@
...
+check_all_tables_populated "Test Case 5"
+(( rc=rc+$? ))
+
+check_all_tables_new_history "Test Case 5"
+(( rc=rc+$? ))
+
+check_return_flag $rc "Test Case 5: Not all tables were populated with inventory data or new history entries were not created."
+
+rm -f ${TEMP_LOG}
+echo "------------------------------------------------------------" >> ${LOG}
```
Comment (pdlun92) [inline csmtest/buckets/basic/node_inventory_db_history.sh]: Removing these dividing lines would also be more consistent with formatting in different buckets
```suggestion
```

**Comment by besawn** [comment]
Pat, I agree with all of your suggestions, and I think all of the updates have been made and committed now. I will wait to merge until you confirm everything looks good to you.

Here is the most recent output when the bucket is successful:
```
------------------------------------------------------------
          Starting Node Inventory DB History Bucket
------------------------------------------------------------
Wed Oct 17 10:28:42 EDT 2018
------------------------------------------------------------
SINGLE_COMPUTE is set to 'c650f99p18'
Configured DB tables for testing: csm_hca csm_gpu csm_node csm_processor_socket csm_dimm csm_ssd
------------------------------------------------------------
Test Case 0: Clean table data; confirm all tables are empty:                                             PASS
Test Case 1: Start the compute daemon; confirm entries are created in the active and history tables:     PASS
Test Case 2: Change the compute node state; confirm no entries are created in the history tables:        PASS
Test Case 3: Restart the compute daemon; confirm no entries are created in the history tables:           PASS
Test Case 4: Stop the compute daemon, delete it from DB; confirm history table entries are created:      PASS
Test Case 5: Start the compute daemon; confirm entries are created in the active and history tables:     PASS
Test Case 6: Change inventory fields and recollect inventory, confirm expected history is created:       PASS
------------------------------------------------------------
             Node Inventory DB History Bucket Passed
------------------------------------------------------------
Failed test cases:

------------------------------------------------------------

```

**Pull request merged**